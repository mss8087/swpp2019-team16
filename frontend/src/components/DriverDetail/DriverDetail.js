import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Button from '../common/Button/Button';

const DriverDetailBlock = styled.div``;

DriverDetail.propTypes = {
  groupId: PropTypes.number.isRequired,
  onClickConfirm: PropTypes.func.isRequired,
};

function DriverDetail({ groupId, onClickConfirm }) {
  const [totalCost, setTotalCost] = useState(0);
  
  return (
    <DriverDetailBlock>
      <input
        type="text"
        id="total-cost-input"
        name="TotalCost"
        placeholder="Total Cost"
        onChange={e => setTotalCost(e.target.value)}
        value={totalCost}
      />

      <Button
        children="Confirm"
        variant="contained"
        fullwidth="false"
        onClick={()=> onClickConfirm({groupId, totalCost})}
      />
      
    </DriverDetailBlock>
  );
}
export default DriverDetail;
