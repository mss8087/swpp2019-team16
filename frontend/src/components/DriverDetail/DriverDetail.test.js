import React from 'react';
import { render } from '@testing-library/react';
import DriverDetail from './DriverDetail';

describe('<DriverDetail />', () => {
  it('SHOULD match with snapshot', async () => { 
    const { container } = render(
      <DriverDetail
        groupId = {1}
        onClickConfirm={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
