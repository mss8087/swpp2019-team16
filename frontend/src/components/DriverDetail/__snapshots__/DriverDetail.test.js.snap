// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DriverDetail /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bDWFJH"
  >
    <input
      id="total-cost-input"
      name="TotalCost"
      placeholder="Total Cost"
      type="text"
      value="0"
    />
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-1 makeStyles-root-4 MuiButton-contained makeStyles-contained-2 makeStyles-contained-5 MuiButton-contained makeStyles-contained-2 makeStyles-contained-5"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Confirm
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
